python/readme.md

```diff
@@ -0,0 +1,196 @@
+```
+
+## Message Databases
+
+In order to decode or encode a message `novatel_edie` requires a database 
+with that message's definition.
+
+For convience `novatel_edie` comes with a built-in message database
+(stored within the `messages.json` file), 
+whose messages and enum values can be accessed via the 
+`novatel_edie.messages` and `novatel_edie.enums` submodules respectively.
+
+```python
+from novatel_edie.messages import BESTPOS
+from novatel_edie.enums import SolStatus
+```
+
+
+If the default message database is inconsistent with message data you want to 
+parse, you can install a different database from a `.json` file by 
+providing it as an argument to the novatel edie command line tool:
+
+```
+novatel_edie install-custom <path-to-your-json-database-file>
+```
+
+In cases where multiple or fully dynamic databases are required, 
+`MessageDatabase` objects can be constructed specified as the source 
+for decoding/encoding processes.
+
+```python
+# Load a fully dynamic database
+custom_db = MessageDatabase('path/to/my/database')
+
+# Access values of the database
+bestpos_type = custom_db.get_message_type('BESTPOS')
+sol_status_enum = db.get_enum_type_by_name('SolStatus')
+
+# Parse based on different databases
+default_db_parser = Parser()
+custom_db_parser = Parser(message_db=custom_db)
+
+```
+
+In general it is recommend to use the default or custom installed database
+whenever possible to take advantage of additional static type information (e.g. autocomplete).
+
+## High Level Parsing
+
+`novatel_edie` provides two classes for parsing message collections,
+a `FileParser` for parsing data stored within a file, 
+and a `Parser` for parsing generic bytes data.
+The two classes share a largely common interface, with the main differences
+being how data is fed into each one.
+
+Data is written to a `Parser` as `bytes` via its `write()` method:
+
+```python
+parser = Parser()
+parser.write(b'message_data')
+```
+
+A `FileParser` receives its data upon instantiation:
+
+```python
+file_parser = FileParser('path/to/log/file')
+```
+
+The simplest way to get information out of either object is by iterating through it. 
+It is recommended to handle different values is by checking their type. 
+You can expect anything returned via iterating through a parser to either 
+be a `Message` or `bytes` object.
+```python
+# Iterate through all messages in a parser
+for msg in parser:
+    if isinstance(msg, bytes):
+        pass
+    elif isinstance(msg, IncompleteMessage):
+        print(msg.header)
+    elif isinstance(msg, BESTPOS):
+        print(msg.latitude)
+```
+Under certain circumstances parsing may encounter an error,
+if you wish to handle this it is recommended to iterate through 
+the `FileParser`/`Parser` object manually via the `read()` method.
+```python
+buffer_size = MAX_BINARY_MESSAGE_LENGTH
+while True:
+    try:
+        msg = parser.read()
+    except Exception as e:
+        logging.exception('An exception occurred', e)
```

Comment:
I think an example of a realistic exception raised by EDIE would be useful here.